images/snapshot-release-tags/prune_snapshots.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import subprocess
+
+import sentry_sdk
+from github import Github
+
+
+sentry_sdk.init(
+    # This cron job only runs once weekly,
+    # so just record all transactions.
+    traces_sample_rate=1.0,
+)
+
+TAG_REF_REGEX = re.compile(r"^refs/tags/(develop-\d{4}-\d{2}-\d{2})$")
+
+
+def main():
+    if "GITHUB_TOKEN" not in os.environ:
+        raise Exception("GITHUB_TOKEN environment is not set")
+
+    parser = argparse.ArgumentParser(
+        prog="prune_snapshots.py",
+        description="Prune expired snapshots",
+    )
+
+    parser.add_argument(
+        "-k",
+        "--keep-last-n",
+        type=int,
+        default=8,
+        help="Prune all but most recent --keep-last-n",
+    )
+    parser.add_argument(
+        "-m",
+        "--mirror-root",
+        default="s3://spack-binaries",
+        help=("Root url of mirror where snapshot binaries are mirrored"),
+    )
+
+    args = parser.parse_args()
+
+    keep_n = args.keep_last_n
+    mirror_root_url = args.mirror_root
+
+    # Use the GitHub API to create a tag for this commit of develop.
+    github_token = os.environ.get("GITHUB_TOKEN")
+    py_github = Github(github_token)
+    py_gh_repo = py_github.get_repo("spack/spack", lazy=True)
+
+    # Get a list of all the tags matching the develop snapshot pattern
+    snapshot_tags = py_gh_repo.get_git_matching_refs("tags/develop-")
```

Comment:
The method described below would break this query for mirror names.

If we still want to use tags, we could move deleted tags to hidden refs, something like `refs/archive/develop-*`. The other option is for each prefix listed in the root, match the name to a regex `<prefix>/develop-*/` check for the `index.json` to see what to delete.

I am not sure which method to prefer for this, maybe use both, one for listing tags to remove, the other for deleting mirrors.